packages/plugin-solana/src/providers/token.ts

```diff
@@ -604,21 +604,13 @@ export class TokenProvider {
         }
 
         // Sort pairs by both liquidity and market cap to get the highest one
-        return dexData.pairs.reduce((highestPair, currentPair) => {
-            const currentLiquidity = currentPair.liquidity.usd;
-            const currentMarketCap = currentPair.marketCap;
-            const highestLiquidity = highestPair.liquidity.usd;
-            const highestMarketCap = highestPair.marketCap;
-
-            if (
-                currentLiquidity > highestLiquidity ||
-                (currentLiquidity === highestLiquidity &&
-                    currentMarketCap > highestMarketCap)
-            ) {
-                return currentPair;
+        return dexData.pairs.sort((a, b) => {
```

Comment:
Looks like the same code in two places can we create a function and reuse it?

Reply:
they are being used by different token providers in each chain plugin, right now the only common place is core and maybe not a good place to add it. For now I thinks its fine like this, maybe creating a service for proving tokens would be better place for this

Reply:
Yeah makes sense feel free to just add an issue or TODO comment so it doesn't get lost. LGTM otherwise thanks so much!